openshift/default-catalog-consistency/pkg/extract/extract.go

```diff
@@ -0,0 +1,167 @@
+package extract
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	imgcopy "github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/pkg/compression"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/archive"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+)
+
+// Result represents the extracted OCI content and related context
+type Result struct {
+	Store   oras.ReadOnlyTarget
+	TmpDir  string
+	Cleanup func()
+}
+
+// UnpackImage pulls the image, extracts it to disk, and opens it as an OCI store.
+func UnpackImage(ctx context.Context, imageRef, name string) (res *Result, err error) {
+	tmpDir, err := os.MkdirTemp("", fmt.Sprintf("oci-%s-", name))
+	if err != nil {
+		return nil, fmt.Errorf("create temp dir: %w", err)
+	}
+
+	cleanup := func() { os.RemoveAll(tmpDir) }
+
+	if err := pullImage(ctx, imageRef, tmpDir); err != nil {
```

Comment:
since we're cleaning up at every error, it could be worth to do something like:

```
store, err := func() {}()
if err != nil {
   os.RemoveAll(tmpDir)
   return err
}
```

just a thought - nothing that needs to be actioned

Reply:
👍 refactored . Good catcher